packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.internal.js

```diff
@@ -972,6 +973,92 @@ describe('ReactHooksWithNoopRenderer', () => {
       },
     );
 
+    it('defers passive effect destroy functions during unmount', () => {
+      function Child({count}) {
+        React.useEffect(() => {
+          Scheduler.unstable_yieldValue('passive create (no dependencies)');
+          return () => {
+            Scheduler.unstable_yieldValue('passive destroy (no dependencies)');
+          };
+        }, []);
+        React.useLayoutEffect(() => {
+          Scheduler.unstable_yieldValue('layout create (no dependencies)');
+          return () => {
+            Scheduler.unstable_yieldValue('layout destroy (no dependencies)');
+          };
+        }, []);
+        React.useEffect(() => {
+          Scheduler.unstable_yieldValue('passive create');
+          return () => {
+            Scheduler.unstable_yieldValue('passive destroy');
+          };
+        }, [count]);
+        React.useLayoutEffect(() => {
+          Scheduler.unstable_yieldValue('layout create');
+          return () => {
+            Scheduler.unstable_yieldValue('layout destroy');
+          };
+        }, [count]);
+        Scheduler.unstable_yieldValue('render');
+        return null;
+      }
+
+      act(() => {
+        ReactNoop.render(<Child count={1} />, () =>
+          Scheduler.unstable_yieldValue('Sync effect'),
+        );
+        expect(Scheduler).toFlushAndYieldThrough([
+          'render',
+          'layout create (no dependencies)',
+          'layout create',
+          'Sync effect',
+        ]);
+        // Effects are deferred until after the commit
+        expect(Scheduler).toFlushAndYield([
+          'passive create (no dependencies)',
+          'passive create',
+        ]);
+      });
+
+      // HACK
+      // This update is kind of gross since it exists to test an internal implementation detail:
+      // Effects without updating dependencies lose their layout/passive tag during an update.
+      // A special type of no-update tag (NoEffectPassiveUnmountFiber) is used to track these for later.
```

Comment:
To make it less "hacky" you could include the count in the log message. So that the effect actually depends on it.

Then make the other effect close over a different prop. `count2` or whatever.

Then during the update, you change one of the props but not the other. Only that effect should fire; the other one will bail out.

Reply:
Yeah, sure. That"hacky" part is that I'm only doing this intermediate update because I *know* that used to cause the specific implementation to blur the passive and layout effects together.